### PR TITLE
Fix manager to not delete and set identical voice commands

### LINKED ISSUE #433: Avoid deleting and setting identical voice commands
### Bug Report
Currently, setting voice commands doesn't check for a diff of which voice commands are in the old and new lists. This results in a case where if voice commands are set like so:

```
voiceCommands = [command1, command2, command3]
voiceCommands = [command1, command2, command3, command 4]
```

What will happen is `command1/2/3` will be set to the head unit, then `command1/2/3` will be deleted from the head unit, then `command1/2/3/4` will be set to the head unit.

Because voice commands can be so slow to process, we should diff and only set / delete what is necessary instead of taking an all-or-nothing approach.

### PR DESCRIPTION
**Pull request by renonick87**
Fixes #433 

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [X] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [X] I have verified that this PR passes lint validation
- [X] I have run the unit tests with this PR
- [X] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
Added unit tests to verify that VoiceCommands are added/deleted correctly.

##### Bug Fixes
* The VoiceCommandManager will no longer delete a VoiceCommand if the VoiceCommand is going to be immediately added again as part of the same update operation. Below is sample code that can be used to test the functionality. Option 1-3 will be added as part of the first operation. The second operation will delete Option 3 and add Option 4 while leaving Option 1/2 untouched.
```
const voiceCommands = [
    new SDL.manager.screen.utils.VoiceCommand(['Option 1'], () => {}),
    new SDL.manager.screen.utils.VoiceCommand(['Option 2'], () => {}),
    new SDL.manager.screen.utils.VoiceCommand(['Option 3'], () => {}),
];

await screenManager.setVoiceCommands(voiceCommands);
const newVoiceCommands = [
    voiceCommands[0],
    voiceCommands[1],
    new SDL.manager.screen.utils.VoiceCommand(['Option 4'], () => {}),
]
await screenManager.setVoiceCommands(newVoiceCommands);
```

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: lib/js/src/manager/screen/utils/_VoiceCommandUpdateOperation.js [hunk 1]
```diff
@@ -82,18 +82,24 @@ class _VoiceCommandUpdateOperation extends _Task {
         this.onFinished();
     }
 
+    _setOldVoiceCommands (oldVoiceCommands) {
```
Comment (crokita) [inline lib/js/src/manager/screen/utils/_VoiceCommandUpdateOperation.js:85]: This doesn't look like it's used anywhere

**Review by crokita — COMMENTED** [review]
See minor feedback

**Review by crokita — APPROVED** [review]

**Comment by renonick87** [comment]
Updates made since last review:
- Equality check on voice commands is not done at the start.
- Added a clone method to the VoiceCommand class and unit tests to cover it.
- The manager's _voiceCommands is set to null if the commands are invalid.
- Updated the docs description for getVoiceCommands since it says that it returns the currently set voice commands but does not actually return _currentVoiceCommands.
- Reduced the logic in _voiceCommandsNotInSecondArray to be much cleaner.
- Fixed the case where updating a Voice Command's listener and calling setVoiceCommands would not update the listener.

File: lib/js/src/manager/screen/utils/_VoiceCommandUpdateOperation.js [hunk 2]
```diff
@@ -119,6 +135,31 @@ class _VoiceCommandUpdateOperation extends _Task {
...
 
+    /**
+     * Returns an array of VoiceCommands that are in the first array but not the second array
+     * @param {VoiceCommmand[]} firstArray - an array of VoiceCommands
+     * @param {VoiceCommand[]} secondArray - an array of VoiceCommands
+     * @returns {VoiceCommand[]} - An array of VoiceCommands that are in the first array but not the second array
+     */
+    _voiceCommandsNotInSecondArray (firstArray = null, secondArray = null) {
+        if (!Array.isArray(firstArray) || firstArray.length === 0) {
+            return secondArray;
```
Comment (crokita) [inline lib/js/src/manager/screen/utils/_VoiceCommandUpdateOperation.js:146]: Elements from the second array should not be in the result according to the docs. Should return `[]` in this case
```suggestion
            return [];
```

**Review by crokita — CHANGES_REQUESTED** [review]
See feedback

**Review by crokita — APPROVED** [review]

**Pull request merged**